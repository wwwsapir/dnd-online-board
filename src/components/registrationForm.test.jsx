import RegistrationForm from "./RegistrationForm";
import { MemoryRouter } from "react-router";
import "mutationobserver-shim";

it("renders correctly", () => {
  const wrapper = shallow(<RegistrationForm onRegistered={() => {}} />);

  expect(wrapper).toMatchSnapshot();
});

jest.mock("../services/registerNewUser");
it("register button functionality - success (async)", async (done) => {
  const spyOnRegistered = sinon.spy();
  const wrapper = mount(
    <MemoryRouter>
      <RegistrationForm onRegistered={spyOnRegistered} />
    </MemoryRouter>
  );
  const userName = wrapper.find("input").first();
  userName.getDOMNode().value = "wwwsapir";
  userName.getDOMNode().dispatchEvent(new Event("input"));
  const email = wrapper.find("input").at(1);
  email.getDOMNode().value = "wwwsapir@gmail.com";
  email.getDOMNode().dispatchEvent(new Event("input"));
  const password = wrapper.find("input").at(2);
  password.getDOMNode().value = "111111";
  password.getDOMNode().dispatchEvent(new Event("input"));

  wrapper.find("button").first().simulate("submit");
  setTimeout(() => {
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    expect(spyOnRegistered.calledOnce).toBe(true);
    done();
  });
});

it("register button functionality - failure (async)", async (done) => {
  const spyOnRegistered = sinon.spy();
  const wrapper = mount(
    <MemoryRouter>
      <RegistrationForm onRegistered={spyOnRegistered} />
    </MemoryRouter>
  );
  const userName = wrapper.find("input").first();
  userName.getDOMNode().value = "other";
  userName.getDOMNode().dispatchEvent(new Event("input"));
  const email = wrapper.find("input").at(1);
  email.getDOMNode().value = "other@gmail.com";
  email.getDOMNode().dispatchEvent(new Event("input"));
  const password = wrapper.find("input").at(2);
  password.getDOMNode().value = "111111";
  password.getDOMNode().dispatchEvent(new Event("input"));

  wrapper.find("button").first().simulate("submit");
  setTimeout(() => {
    wrapper.update();
    expect(wrapper).toMatchSnapshot();
    expect(spyOnRegistered.called).toBe(false);
    done();
  });
});
