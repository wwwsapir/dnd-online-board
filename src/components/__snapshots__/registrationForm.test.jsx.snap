// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`register button functionality - failure (async) 1`] = `
<div
  className="menu-window"
>
  <form
    onSubmit={[Function]}
  >
    <ul
      className="menu bg-dark w-100"
    >
      <h4
        className="mb-5"
      >
        <span
          className="menu-header"
        >
          New User Registration
        </span>
      </h4>
      <li
        className="mb-3"
      >
        <input
          className="input-group-sm form-control"
          name="userName"
          placeholder="User Name"
          type="text"
        />
        <label
          className="error badge badge-danger"
        >
          User name length should be 6-72 characters
        </label>
      </li>
      <li
        className="mb-3"
      >
        <input
          className="input-group-sm form-control"
          name="email"
          placeholder="Email Address"
          type="email"
        />
        <label
          className="error badge badge-danger"
        />
      </li>
      <li>
        <input
          className="input-group-sm form-control mt-3"
          name="password"
          placeholder="Password"
          type="password"
        />
        <label
          className="error badge badge-danger"
        />
      </li>
      <li>
        <button
          className="btn btn-primary form-control mt-3"
          disabled={false}
          type="submit"
        >
          Register
        </button>
      </li>
      <li
        className="mt-2 mb-4"
      >
        <Link
          className="float-right"
          to="/home/login"
        >
          <LinkAnchor
            className="float-right"
            href="/home/login"
            navigate={[Function]}
          >
            <a
              className="float-right"
              href="/home/login"
              onClick={[Function]}
            >
              Already registered? log in
            </a>
          </LinkAnchor>
        </Link>
      </li>
    </ul>
  </form>
</div>
`;

exports[`register button functionality - success (async) 1`] = `""`;

exports[`renders correctly 1`] = `
<div
  className="menu-window"
>
  <form
    onSubmit={[Function]}
  >
    <ul
      className="menu bg-dark w-100"
    >
      <h4
        className="mb-5"
      >
        <span
          className="menu-header"
        >
          New User Registration
        </span>
      </h4>
      <li
        className="mb-3"
      >
        <input
          className="input-group-sm form-control"
          name="userName"
          placeholder="User Name"
          type="text"
        />
        <label
          className="error badge badge-danger"
        />
      </li>
      <li
        className="mb-3"
      >
        <input
          className="input-group-sm form-control"
          name="email"
          placeholder="Email Address"
          type="email"
        />
        <label
          className="error badge badge-danger"
        />
      </li>
      <li>
        <input
          className="input-group-sm form-control mt-3"
          name="password"
          placeholder="Password"
          type="password"
        />
        <label
          className="error badge badge-danger"
        />
      </li>
      <li>
        <button
          className="btn btn-primary form-control mt-3"
          disabled={false}
          type="submit"
        >
          Register
        </button>
      </li>
      <li
        className="mt-2 mb-4"
      >
        <Link
          className="float-right"
          to="/home/login"
        >
          Already registered? log in
        </Link>
      </li>
    </ul>
  </form>
</div>
`;
